packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -253,6 +313,36 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
       },
     })
 
+  useEffect(() => {
+    // When changes are made (but not saved) to the datasetToUpdate, show a notification that summarizes the change and prompts the user to save
+    if (
+      previousDatasetToUpdate &&
+      datasetToUpdate &&
+      !isEqual(previousDatasetToUpdate.items, datasetToUpdate.items) &&
+      !updateIsPending
+    ) {
+      const toastMessageTitle = getDatasetChangedToastMessageTitle(
+        previousDatasetToUpdate,
+        datasetToUpdate,
+      )
+      // Replace existing toast that prompts the user to save before showing the new one
+      if (onCloseLastSavePromptToast) {
+        onCloseLastSavePromptToast.close()
+      }
+      const onCloseToast = displayToast(SAVE_TO_CONTINUE, 'info', {
+        title: toastMessageTitle,
+        primaryButtonConfig: {
+          text: SAVE_CHANGES,
+          onClick: () => mutate(datasetToUpdate),
+        },
+      })
+      setOnCloseLastSavePromptToast({ close: onCloseToast })
```

Comment:
Remove the last toast and update the state variable so this one can be closed in the same fashion